Post: Extending material theme in Jetpack Compose.

mirland opened the pull request:
TODO:
- [x] Add links
- [x] Add a banner image
- [x] Code review
- [x] Change blog date

Blog URL https://blog-staging.xmartlabs.com/blog/extending-material-theme-in-jetpack-compose/

sante commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+If you are a Jetpack Compose user you may already know that one of its advantages is that it's easier to give your app a consistent look & feel by applying themes.
+Additionally, Material already provides a theme that allows apps based on Material Design to reflect your own product’s branding and styles.
+And that is great, but it oftentimes lacks in that it's a bit too strict and limited on what you can and cannot do with it.
+In this post, we'll analyze how to adapt, extend and make more flexible the Material theme so that it can stick to your product's style guidelines.
+​
```
```suggestion
In this post, we'll analyze how to adapt, extend and make Metrial Theme more flexible so that it can stick to your product's style guidelines.
​
```

sante commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+Themes are a collection of named resources that are useful all throughout a given application.
+More information can be found in [Android's themes documentation], but themes assign semantic names like `colorPrimary` to Android resources, that can be later used as references in different places on the app.
+​
+Most Android applications already follow (or should be following) [Material Design Guidelines], and that's why Material created the `MaterialTheme`.
```
```suggestion
Most Android applications already follow (or should be following) [Material Design Guidelines], and that's why Material created `MaterialTheme`.
```

sante commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+​
+# Conclusions
+UI/UX is one of the most important things in a mobile app.
+Material has some guides to help improve that and it also allows to have consistency wit other apps on the same platform.
```
```suggestion
Material has some guides to help improve that and it also allows to have consistency with other apps on the same platform.
```

sante commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+# Conclusions
+UI/UX is one of the most important things in a mobile app.
+Material has some guides to help improve that and it also allows to have consistency wit other apps on the same platform.
+Themes allow for consistency throughout our own application, especially if you define it at the beginning, just by applying minor configurations.
```
```suggestion
Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.
```

sante commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+UI/UX is one of the most important things in a mobile app.
+Material has some guides to help improve that and it also allows to have consistency wit other apps on the same platform.
+Themes allow for consistency throughout our own application, especially if you define it at the beginning, just by applying minor configurations.
+However, we saw that the Material theme is not really all that flexible, and if designers don't stick 100% to the their guidelines, you may be in trouble as you may now have a mix of Material and other theme not really related to it.
```
```suggestion
However, we saw that Material theme is not really all that flexible, and if designers don't stick 100% to the their guidelines, you may be in trouble as you may now have a mix of Material and other theme not really related to it.
```

sante commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+Themes allow for consistency throughout our own application, especially if you define it at the beginning, just by applying minor configurations.
+However, we saw that the Material theme is not really all that flexible, and if designers don't stick 100% to the their guidelines, you may be in trouble as you may now have a mix of Material and other theme not really related to it.
+In this post we presented some ideas to avoid these issues, a way to extend the Material theme while making it flexible, and the most important thing, adaptable to your product.
+[The code in this blog should be useful for most application, I recommend you check the full implementation on GitHub], and adapt the general ideas to your specific use case.
```
```suggestion
[The code in this blog should be useful for most applications, but I recommend you check out the full implementation on GitHub], and adapt the general ideas to your specific use case.
```

mirland commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+author_id: mirland
+show: true
+featured_image: /images/recap2020/look-back-2020-featured.jpg
+permalink: /blog/extending-material-them-in-jetpack-compose/
```
```suggestion
permalink: /blog/extending-material-theme-in-jetpack-compose/
```

f7deleon commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+---
+​
+If you are a Jetpack Compose user you may already know that one of its advantages is that it's easier to give your app a consistent look & feel by applying themes.
+Additionally, Material already provides a theme that allows apps based on Material Design to reflect your own product’s branding and styles.
```
maybe avoid saying material two times in the same sentence
```suggestion
Additionally, Material already provides a theme that allows apps based on it to reflect your own product’s branding and styles.
```

f7deleon commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+UI/UX is one of the most important things in a mobile app.
+Material has some guides to help improve that and it also allows to have consistency with other apps on the same platform.
+Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.
+However, we saw that Material theme is not really all that flexible, and if designers don't stick 100% to the their guidelines, you may be in trouble as you may now have a mix of Material and other theme not really related to it.
```
```suggestion
However, we saw that Material theme lacks flexibility, and if designers don't stick 100% to the their guidelines, you may be in trouble as you may now have a mix of Material and other theme not really related to it.
```

f7deleon commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.
+However, we saw that Material theme is not really all that flexible, and if designers don't stick 100% to the their guidelines, you may be in trouble as you may now have a mix of Material and other theme not really related to it.
+In this post we presented some ideas to avoid these issues, a way to extend the Material theme while making it flexible, and the most important thing, adaptable to your product.
+[The code in this blog should be useful for most applications, but I recommend you check out the full implementation on GitHub], and adapt the general ideas to your specific use case.
```
maybe change this link to just github?
```suggestion
The code in this blog should be useful for most applications, but I recommend you check out the full implementation on [GitHub], and adapt the general ideas to your specific use case.
```

f7deleon commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+​
+# What's an application theme?
+The definition is not complex.
+Themes are a collection of named resources that are useful all throughout a given application.
```
Maybe omitting the named resources since in the next line you explain that its assigns names to resources?
```suggestion
Themes are a collection of resources that are useful all throughout a given application.
```

f7deleon commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+​
+​
+### Sounds good, but is it perfect?
+Material does a great job, but from my point of view, it's too strict.
```
a suggestions both ways are ok i think
```suggestion
Material does a great job, but from my point of view, it lacks flexibility.
```

bryant1410 commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+---
+layout: post
+title: Extending Material theme in Jetpack Compose.
```
```suggestion
title: Extending Material theme in Jetpack Compose
```

bryant1410 commented on Gemfile.lock:
```diff
@@ -77,11 +77,12 @@ GEM
 
 PLATFORMS
   ruby
+  x86_64-linux-musl
```
Just trying to understand, why this?

mirland replied:
I can remove it, I tested to update some dependencies and it was added, but I have no idea. I'll ask bala

mirland replied:
By bala https://www.moncefbelyamani.com/understanding-the-gemfile-lock-file/#platforms

bryant1410 commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+# What's an application theme?
+The definition is not complex.
+Themes are a collection of resources that are useful throughout a given application.
+More information can be found in [Android's themes documentation], but themes assign semantic names like `colorPrimary` to Android resources, that can be later used as references in different places on the app.
```
```suggestion
More information can be found in [Android's themes documentation], but themes assign semantic names such as `colorPrimary` to Android resources, that can be later used as references in different places on the app.
```

"such as" should go for real examples. "like" is more like for figurative or related things.

bryant1410 reviewed (commented):
I send some minor comments so you can process them while I'm reading!

mirland commented on docker-compose.yml:
```diff
@@ -1,7 +1,7 @@
 services:
   jekyll:
-    image: jekyll/jekyll:3.9
+    image: jekyll/jekyll:3.8
```
3.9 doesn't exist, so it's not working (https://hub.docker.com/r/jekyll/jekyll/tags?page=1&ordering=last_updated)

bryant1410 reviewed (changes requested):
It's hard to judge the content a lot as I'm not doing Android development these days, but it does sound good!

Important things to change:
* I'd give more context at the very beginning. Mentioning some keywords such as "Android", "mobile development", and maybe others. The rationale is the following. If you're very familiar with the things mentioned in this post, you won't struggle. However, if you're familiar with some of them (or not familiar at all), it's hard to tell what's the context of this post (e.g., Android or iOS? is it about mobile development? is it about the web?).
* Paragraphs are broken on blog-staging, for whatever reason.
* Why did you remove your author profile pic?

Less important things:
* Use contractions as much as you can. My understanding is that the blog post tone is informal. Not using contractions is quite formal.
* In many cases here I wouldn't use the future tense ("will"). Maybe revise it and change it to present tense.

Left many comments mostly regarding the wording, to make it clearer.

bryant1410 commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+permalink: /blog/extending-material-theme-in-jetpack-compose/
+---
+​
+If you are a Jetpack Compose user you may already know that one of its advantages is that it's easier to give your app a consistent look & feel by applying themes.
```
Also for context, what about adding some reference link here to Jetpack Compose?

mirland replied:
What about something like:

```suggestion
If you have tried [Jetpack Compose], the new Android's modern toolkit for building UIs, you may already know that one of its advantages is that it's easier to give your app a consistent look & feel by applying themes.
```

?

In a few words, it add context and some documentation

bryant1410 replied:
I like it!

bryant1410 replied:
What about "building UIs" -> "building mobile user interfaces"? As it's the very first sentence.

bryant1410 replied:
Oh, and having both "new" and "modern" is redundant.

mirland replied:
> building mobile user interfaces"

It's the new Android's toolkit, but it's not just jor android, it's unbundled, it's available on desktop and web (both are on alpha).
So, maybe just:

```suggestion
If you have tried [Jetpack Compose], the new Android's toolkit for building UIs, you may already know that one of its advantages is that it's easier to give your app a consistent look & feel by applying themes.
```
My point here, it applies to web and desktop, but anybody should be using it

bryant1410 replied:
maybe: "the new framework for building user interfaces for Android ([and beta for Web and Desktop]), you may ...", with a link ?

bryant1410 replied:
Also: "you may already know that one of its advantages is that it's easier to give your app a consistent look & feel by applying themes." -> "you may know it's now easier to ..."

And I personally don't like the ampersand there, I'd use and. But I'm not 100% sure.

mirland commented:
> It's hard to judge the content a lot as I'm not doing Android development these days, but it does sound good!

Thanks For the review! 

> Important things to change:

> I'd give more context at the very beginning. Mentioning some keywords such as "Android", "mobile development", and maybe others. The rationale is the following. If you're very familiar with the things mentioned in this post, you won't struggle. However, if you're familiar with some of them (or not familiar at all), it's hard to tell what's the context of this post (e.g., Android or iOS? is it about mobile development? is it about the web?).

It makes totally sense

> Paragraphs are broken on blog-staging, for whatever reason.
Good catch, I will check it

> Why did you remove your author profile pic?
I removed bytes, I've not removed my pic. I changed it to be consistent with xl brand

> Less important things:
Use contractions as much as you can. My understanding is that the blog post tone is informal. Not using contractions is quite formal.
In many cases here I wouldn't use the future tense ("will"). Maybe revise it and change it to present tense.
Left many comments mostly regarding the wording, to make it clearer.

:ok_hand:

bryant1410 commented:
> I removed bytes, I've not removed my pic. I changed it to be consistent with xl brand

Oh, I see.

And again, I'd shrink the code font size by about 2 pixels.

mirland commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+UI/UX is one of the most important things in a mobile app.
+Material has some guides to help improve that and it also allows to have consistency with other apps on the same platform.
+Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.
+However, we saw that Material theme lacks flexibility, and if designers don't stick 100% to their guidelines, you may be in trouble as you may now have a mix of Material and another theme not related to it.
```
What this phrase try to tell is: if you use the material theme without these recommendations, you may have the material theme and a lot of separate theme components (that are not handled by your app theme) in your app. You have to mix both and it will be hard to maintain (and the code may be horrible).

> you may be in trouble as you will have to maintain multiple themes.

Your app theme will be divided into multiple themes (or worst, only the material theme and a lot of separate components in your app). However, is it clear?

bryant1410 replied:
Yeah. Maybe say "as you will have to maintain many separate theme components not handled directly by your app theme"?

mirland commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+​
+To extend these classes, we can follow the same idea, we can create a [`AppShapes`] class and a [`AppTypographies`] class the same way and declare custom theme properties.
+​
+I will not explain the code because it's analog, you can find it on [GitHub](https://github.com/xmartlabs/gong/tree/b0b617e56403c1f499704111acad89093aa3c9d6/app/src/main/java/com/xmartlabs/gong/ui/theme) and if you have any questions, don't forget to post a comment!
```
I can say it's similar to the previous one. 
In this case, what's better, similar, or consistent?

bryant1410 replied:
"similar to the previous one"?

mirland commented on _sass/_base.scss:
```diff
@@ -162,7 +162,7 @@ blockquote {
 pre,
 code {
-  font-size: 15px;
+  font-size: 15px !important;
```
> And again, I'd shrink the code font size by about 2 pixels.

I changed 3 pixels, it was the previous value before the website redesign, what do you think?

I've also changed it in the webflow css file, but I think we'll lose this change in the future

bryant1410 replied:
It looks better!

bryant1410 commented on _posts/2021-03-07-compose-themes.md:
````diff
@@ -0,0 +1,236 @@
+      AppColorPalette.BLUE -> if (darkTheme) darkBlueColors else lightBlueColors
+      AppColorPalette.PINK -> if (darkTheme) darkPinkColors else lightPinkColors
+    }
+
+```
````
```suggestion
```
```

So the rendered code block just fits the code :smile:

bryant1410 commented on _posts/2021-03-07-compose-themes.md:
````diff
@@ -0,0 +1,236 @@
+    val listItemVerticalPadding: Dp,
+    val listItemHorizontalPadding: Dp,
+)
+
+```
````
Remove this new line so the rendered code block just fits the text :sweat_smile:

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+permalink: /blog/extending-material-theme-in-jetpack-compose/
+---
+
+If you have tried [Jetpack Compose], the new framework for building user interfaces for Android (and in early steps for [Desktop](https://www.jetbrains.com/lp/compose/) and [Web](https://blog.jetbrains.com/kotlin/2021/05/technology-preview-jetpack-compose-for-web/)), you may know it's now easier to give your app a consistent look and feel by applying themes.
```
```suggestion
Building Android UIs with a consistent look and feel using the new [Jetpack Compose] framework it's easier than ever by setting up themes.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+---
+
+If you have tried [Jetpack Compose], the new framework for building user interfaces for Android (and in early steps for [Desktop](https://www.jetbrains.com/lp/compose/) and [Web](https://blog.jetbrains.com/kotlin/2021/05/technology-preview-jetpack-compose-for-web/)), you may know it's now easier to give your app a consistent look and feel by applying themes.
+Additionally, Material already provides a theme that allows apps based on it to reflect your own product's branding and styles.
+If you haven't tried it, I'd recommend you to check out [Android's documentation](https://developer.android.com/jetpack/compose/themes), it's very simple and clear how you can use it in your app.
+Material theme is great, but it oftentimes lacks in that it's a bit too strict and limited on what you can and cannot do with it.
```
```suggestion
Compose offers an implementation of the Material design that makes it easy to use Material style throughout the whole app just by adding few lines of code. As you may already know, Material theme allows us to have a consistent and uniform UI that follows Android UI best practices and guidelines, but at the same time, it's super rigid and limited when the app needs a more complex UI customization level.
```

mirland replied:
I don't like the idea of losing the info that here we will not explain how to use the material theme, it's something that you should read before this blog. I think it's important

mirland replied:
```suggestion
Compose offers an implementation of the Material design that makes it easy to use Material style throughout the whole app just by adding few lines of code.
If you haven't tried it, you can check out [Android's documentation](https://developer.android.com/jetpack/compose/themes).
As you may already know, Material theme allows us to have a consistent and uniform UI that follows Android UI best practices and guidelines, but at the same time, it's super rigid and limited when the app needs a more complex UI customization level.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+If you haven't tried it, I'd recommend you to check out [Android's documentation](https://developer.android.com/jetpack/compose/themes), it's very simple and clear how you can use it in your app.
+Material theme is great, but it oftentimes lacks in that it's a bit too strict and limited on what you can and cannot do with it.
+
+In this post, we'll not talk about how can you apply the Material theme in your app, we'll analyze how to adapt, customize, extend and make Material Theme more flexible so that it can stick to your product's style guidelines.
```
```suggestion
In this post, we'll learn how to adapt, customize and extend Compose Material Theme in order to fulfill your app UI requirement and get rid of material limitations. 
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+In this post, we'll not talk about how can you apply the Material theme in your app, we'll analyze how to adapt, customize, extend and make Material Theme more flexible so that it can stick to your product's style guidelines.
+
+# What's an application theme?
+The definition is not complex.
```
```suggestion
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+
+# What's an application theme?
+The definition is not complex.
+Themes are a collection of resources that are useful throughout a given application.
+More information can be found in [Android's themes documentation], but themes assign semantic names such as `colorPrimary` to Android resources, that can be later used as references in different places on the app.
```
```suggestion
[Android Theme] is just a collection of resources that are useful throughout a given application. Each resource has a semantic name such as `colorPrimary` that can be used as references from different places on the app.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+Themes are a collection of resources that are useful throughout a given application.
+More information can be found in [Android's themes documentation], but themes assign semantic names such as `colorPrimary` to Android resources, that can be later used as references in different places on the app.
+
+Most Android applications already follow (or should be following) [Material Design Guidelines], and that's why Material created `MaterialTheme`.
```
```suggestion
Most Android apps already follow [Material Design Guidelines], and that's the reason Android SDK already provides the `MaterialTheme` for Compose apps.
```

mirland replied:
it's not the android sdk... it super confusing 
```suggestion
Most Android apps already follow [Material Design Guidelines], and that's the reason Jetpack Compose already provides the `MaterialTheme` for Compose apps.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+Material Theme comprises [color], [typography], and [shape] attributes.
+
+### Sounds good, but is it perfect?
+Material does a great job, but from my point of view, it lacks flexibility.
```
```suggestion
Material does a great job making sure every app follows android UI guidelines, but it clearly lacks flexibility.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+
+### Sounds good, but is it perfect?
+Material does a great job, but from my point of view, it lacks flexibility.
+So, here lies the **first problem**: you have to adapt your look and feel to the Material guidelines, and if you don't, then it becomes hard to use it.
```
So, here lies the **first problem**: Material theme by itself it's not flexible enough to build complex app look and feels.

mirland replied:
if we don't use bold in the next sentence, we should remove it here too
```suggestion
So, here lies the first problem: you have to adapt your look and feel to the Material guidelines, and if you don't, then it becomes hard to use it.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+### Sounds good, but is it perfect?
+Material does a great job, but from my point of view, it lacks flexibility.
+So, here lies the **first problem**: you have to adapt your look and feel to the Material guidelines, and if you don't, then it becomes hard to use it.
+The **second issue** is that Material comprises only [color], [typography], and [shape] attributes, but what if we want to define more resources in a theme, such as dimensions or icons?
```
```suggestion
Secondly, Material comprises only [color], [typography], and [shape] attributes, but what if your app needs more resources, such as dimensions or icons?
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+The **second issue** is that Material comprises only [color], [typography], and [shape] attributes, but what if we want to define more resources in a theme, such as dimensions or icons?
+
+# Extending Material Colors
+Compose provides the [`Colors`] class to model the [Material color system].
+It's a class that defines the 9 main Material colors, and it also provides two functions to define both the light and dark color sets.
```
```suggestion
Compose provides the [`Colors`] class to model the [Material color system] which includes nine colors and two functions to define light and dark color sets.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+Compose provides the [`Colors`] class to model the [Material color system].
+It's a class that defines the 9 main Material colors, and it also provides two functions to define both the light and dark color sets.
+
+In my experience, 9 colors are not enough, as design teams usually define many more.
```
```suggestion
According to our experience, most apps need way more than 9 colors.
```

mtnbarreto replied:
El `as design teams usually define many more.` es un detalle que se puede omitir, en mi opinipn no aporta nada.

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+# Adding custom resources to your theme
+
+In the previous section, we learned how to extend the properties provided by Material, but what if we wanted to define new ones?
+Remember, a theme is a semantic resource set, so we could also have dimensions, icons, or any other resource that will be used in the app.
```
```suggestion
Remember, a theme is a semantic resource set, so we could also need dimensions, icons, or any other resource to build the custom app UI.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+
+## Defining dimensions in our theme
+
+I usually define dimensions in my theme since many of them have a meaning for me.
```
No me gusta el uso de `I`, por momentos es poco inclusivo con el equipo.

mirland replied:
maybe: 
```suggestion
Dimensions are something that we usually define in our theme since many of them have a meaning in our app context.
```

mtnbarreto commented on _posts/2021-03-07-compose-themes.md:
```diff
@@ -0,0 +1,233 @@
+
+# Adding custom resources to your theme
+
+In the previous section, we learned how to extend the properties provided by Material, but what if we wanted to define new ones?
```
```suggestion
So far, we've learned how to extend the properties provided by Material, but what if we wanted to define new ones?
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+---
+
+Building Android UIs with a consistent look and feel is easier than ever by using the new [Jetpack Compose] framework and setting up themes.
+Compose offers an implementation of the Material design that makes it easy to use Material style throughout the whole app just by adding few lines of code.
+You can read more about Compose themes at the [official documentation](https://developer.android.com/jetpack/compose/themes).
```
```suggestion
Compose offers an implementation of the Material design that makes it easy to use Material style throughout the whole app. 
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+permalink: /blog/extending-material-theme-in-jetpack-compose/
+---
+
+Building Android UIs with a consistent look and feel is easier than ever by using the new [Jetpack Compose] framework and setting up themes.
```
```suggestion
Building Android UIs with a consistent look and feel is easier than ever by using the new [Jetpack Compose] framework and setting up [themes](https://developer.android.com/jetpack/compose/themes).
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+Building Android UIs with a consistent look and feel is easier than ever by using the new [Jetpack Compose] framework and setting up themes.
+Compose offers an implementation of the Material design that makes it easy to use Material style throughout the whole app just by adding few lines of code.
+You can read more about Compose themes at the [official documentation](https://developer.android.com/jetpack/compose/themes).
+As you may already know, Material theme allows us to have a consistent and uniform UI that follows Android UI best practices and guidelines, but at the same time, it's super rigid and limited when the app needs a more complex UI customization level.
```
```suggestion
As you may already know, Material allows us to have a consistent and uniform UI that follows Android UI best practices and guidelines, but at the same time, it's super rigid and limited when the app needs a more complex UI customization level.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+[Android Theme] is just a collection of resources that are useful throughout a given application.
+Each resource has a semantic name such as `colorPrimary` that can be used as references from different places on the app.
+
+Most Android apps already follow [Material Design Guidelines], and that's the reason Jetpack Compose already provides the `MaterialTheme` for Compose apps.
```
```suggestion
Most Android apps already follow [Material Design Guidelines], and that's the reason Jetpack Compose provides the `MaterialTheme` for Compose apps by default.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+Each resource has a semantic name such as `colorPrimary` that can be used as references from different places on the app.
+
+Most Android apps already follow [Material Design Guidelines], and that's the reason Jetpack Compose already provides the `MaterialTheme` for Compose apps.
+It's a systematic way to customize Material Design to better reflect your product’s design needs.
```
```suggestion
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+Material Theme comprises [color], [typography], and [shape] attributes.
+
+### Sounds good, but is it perfect?
+Material does a great job making sure every app follows android UI guidelines, but it clearly lacks flexibility.
```
```suggestion
Material does a great job making sure every app follows Android UI guidelines, but it clearly lacks flexibility.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+
+### Sounds good, but is it perfect?
+Material does a great job making sure every app follows android UI guidelines, but it clearly lacks flexibility.
+So, here lies the first problem: you have to adapt your look and feel to the Material guidelines, and if you don't, then it becomes hard to use it.
+Secondly, Material comprises only [color], [typography], and [shape] attributes, but what if your app needs more resources, such as dimensions or icons?
```
```suggestion
So, here lies the first problem: you have to adapt your look and feel to the Material guidelines, and if you don't, then it becomes hard to use it.

Secondly, Material comprises only [color], [typography], and [shape] attributes, but what if your app needs more resources, such as dimensions or icons?
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+[Android Theme] is just a collection of resources that are useful throughout a given application.
+Each resource has a semantic name such as `colorPrimary` that can be used as a reference from different places on the app.
+
+Most Android apps already follow [Material Design Guidelines], that's the reason Jetpack Compose provides the `MaterialTheme` implementation for Compose apps.
+It's a systematic way to customize Material Design to better reflect your product’s design needs.
```
```suggestion
Most Android apps already follow [Material Design Guidelines], that's the reason Jetpack Compose provides the `MaterialTheme` implementation for Compose apps, which is a systematic way to customize Material Design to better reflect your app style and branding.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,231 @@
+# Conclusion
+UI/UX is one of the most important aspects of a mobile app.
+Material has some guides to help improve it and also allows you to be consistent with other apps on the same platform.
+Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.
+However, we saw that Material themes lack flexibility, and if designers don't stick 100% to their guidelines, you may be in trouble as you will have to maintain many separate theme components not handled directly by your app theme.
```
```suggestion
Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.

However, we saw that Material themes lack flexibility, and if designers don't stick 100% to their guidelines, you may be in trouble as you will have to maintain many separate theme components not handled directly by your app theme.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
````diff
@@ -0,0 +1,231 @@
+)
+```
+
+That's all fine, but you may be asking yourself a question which is: what are the actual advantages?
+First, your theme will be consistent.
+You can reuse the dimensions in multiple places throughout your app and thus make sure the correct values are enforced.
+Second, you gain flexibility as you can define custom dimensions based on the device's state or specs.
+A practical example, considering devices with small screens, is to define a separate set of dimensions for them to improve the user experience.
+Last but not least, you avoid magic numbers making your codebase more maintainable and readable.
````
```suggestion
That's all fine, but you may be asking yourself a question which is: what are the actual advantages?

First, your theme will be consistent.
You can reuse the dimensions in multiple places throughout your app and thus make sure the correct values are enforced.

Second, you gain flexibility as you can define custom dimensions based on the device's state or specs.
A practical example, considering devices with small screens, is to define a separate set of dimensions for them to improve the user experience.

Last but not least, you avoid magic numbers making your codebase more maintainable and readable.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,235 @@
+Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.
+
+However, we saw that Material themes lack flexibility, and if designers don't stick 100% to their guidelines, you may be in trouble as you will have to maintain many separate theme components not handled directly by your app theme.
+In this post, we teach you how to overcome all these issues by extending the Material theme while making it flexible to adapt it to your product.
+The code in this blog should be useful for most applications, but I recommend you check out [the full implementation on GitHub](https://github.com/xmartlabs/gong/tree/b0b617e/app/src/main/java/com/xmartlabs/gong/ui/theme), and adapt the general ideas to your specific use case.
```
```suggestion

In this post, we teach you how to overcome all these issues by extending the Material theme while making it flexible to adapt it to your product.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,235 @@
+Building Android UIs with a consistent look and feel is easier than ever by using the new [Jetpack Compose] framework and setting up [themes](https://developer.android.com/jetpack/compose/themes).
+Compose offers a Material theme implementation that makes it easy to use Material style throughout the whole app. 
+
+As you may already know, Material allows us to have a consistent and uniform UI that follows Android UI best practices and guidelines, but at the same time, it's super rigid and limited when the app needs a more complex UI customization level.
```
```suggestion
As you may already know, Material allows us to have a uniform UI that follows Android UI best practices and guidelines, but at the same time, it's super rigid and limited when the app needs a more complex UI customization level.
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,235 @@
+
+As you may already know, Material allows us to have a consistent and uniform UI that follows Android UI best practices and guidelines, but at the same time, it's super rigid and limited when the app needs a more complex UI customization level.
+
+In this post, we'll learn how to adapt, customize and extend Compose Material Theme in order to fulfill your app UI requirement and get rid of material limitations. 
```
```suggestion
In this post, we'll learn how to adapt, customize and extend Compose Material Theme in order to fulfill your app UI requirements and get rid of material limitations. 
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,235 @@
+
+# Conclusion
+UI/UX is one of the most important aspects of a mobile app.
+Material has some guides to help improve it and also allows you to be consistent with other apps on the same platform.
```
```suggestion
Material helps in that regard by ensuring every Android app follows Android's UI practices and its overall guidelines. 
```

mtnbarreto commented on _posts/2021-05-07-compose-themes.md:
```diff
@@ -0,0 +1,235 @@
+# Conclusion
+UI/UX is one of the most important aspects of a mobile app.
+Material helps in that regard by ensuring every Android app follows Android's UI practices and its overall guidelines. 
+Themes allow for consistency throughout an application, especially if you define it at the beginning, just by applying minor configurations.
```
```suggestion
Themes allow consistency throughout your application, especially if it's configured at the beginning of the app development.
```

bryant1410 commented on _posts/2021-05-10-compose-themes.md:
```diff
@@ -0,0 +1,239 @@
+Theming the application using Material Theme helps when it comes to encapsulating and reutilizing styles and assets.
+There is nothing worse than having a huge app that is a mess regarding theme and assets management and configuration.
+
+In this blog post, we present concepts and ideas to leverage compose, app theming, and Material and if your app still needs more customization, be able to extend material capabilities.
```
I think it's clearer this way:

```suggestion
This blog post presents concepts and ideas to leverage compose, app theming, and Material. If your app still needs more customization, you'll be able to extend Material capabilities.
```

mirland replied:
Maybe?
```suggestion
This blog post presents concepts and ideas to leverage compose, app theming, and Material.
If your app still needs more customization, you'll be able to extend Material capabilities to derive full benefit from them.
```
